File: _includes/code/python/quickstart.short.local.query.neartext.py
# START NearText
import weaviate
import json

# Step 2.1: Connect to your local Weaviate instance
with weaviate.connect_to_local() as client:

    # Step 2.2: Use this collection
    movies = client.collections.use("Movie")

    # Step 2.3: Perform a semantic search with NearText
    # highlight-start
    # END NearText
    # fmt: off
    # START NearText
    response = movies.query.near_text(
        query="sci-fi",
        limit=2
    )
    # END NearText
    # fmt: on
    # START NearText
    # highlight-end

    for obj in response.objects:
        print(json.dumps(obj.properties, indent=2))  # Inspect the results
    # END NearText

File: _includes/code/howto/go/docs/quickstart/short_local_2/quickstart.short.local.query.neartext.go
package main

// START NearText
import (
	"context"
	"encoding/json"
	"fmt"

	"github.com/weaviate/weaviate-go-client/v5/weaviate"
	"github.com/weaviate/weaviate-go-client/v5/weaviate/graphql"
)

func main() {
	// Step 1.1: Connect to your local Weaviate instance
	cfg := weaviate.Config{
		Host:   "localhost:8080",
		Scheme: "http",
	}
	client, err := weaviate.NewClient(cfg)
	if err != nil {
		panic(err)
	}

	// Step 2.2: Perform a semantic search with NearText
	// highlight-start
	title := graphql.Field{Name: "title"}
	description := graphql.Field{Name: "description"}
	genre := graphql.Field{Name: "genre"}

	nearText := client.GraphQL().NearTextArgBuilder().
		WithConcepts([]string{"sci-fi"})

	result, err := client.GraphQL().Get().
		WithClassName("Movie").
		WithNearText(nearText).
		WithLimit(2).
		WithFields(title, description, genre).
		Do(context.Background())
	// highlight-end

	if err != nil {
		panic(err)
	}

	// Inspect the results
	if result.Errors != nil {
		fmt.Printf("Error: %v\n", result.Errors)
		return
	}

	data := result.Data["Get"].(map[string]interface{})
	movies := data["Movie"].([]interface{})

	for _, movie := range movies {
		jsonData, err := json.MarshalIndent(movie, "", "  ")
		if err != nil {
			panic(err)
		}
		fmt.Println(string(jsonData))
	}
	// END NearText
}
